dali/kernels/slice/slice_gpu.cuh

```diff
@@ -72,9 +72,28 @@ struct SliceBlockDesc {
   uint64_t size;
 };
 
-template<typename OutputType>
-constexpr int coalesced_values = sizeof(OutputType) >= 4 ? 1 : 4 / sizeof(OutputType);
-
+template<typename T>
+union PackedBuffer {
+  using PackedType = uint32_t;
+  static constexpr size_t kCapacity = sizeof(T) >= sizeof(PackedType) ?
+                                      1 : sizeof(PackedType) / sizeof(T);
+
+  T values[kCapacity];
+  PackedType raw;
+
+  __device__ inline void store(T* mem, size_t count) {
+    if (kCapacity == 1) {
+      *mem = *values;
+    } else if (count == kCapacity && (uint64_t)mem % sizeof(PackedType) == 0) {
```

Comment:
```suggestion
    } else if (count == kCapacity && (uintptr_t)mem % sizeof(PackedType) == 0) {
```

Reply:
I guess that `std::align` would be an overkill here.

Reply:
I've applied your suggestion, thanks! I also think that `std::align` would be an overkill, especially for this performance-critical part of code

Reply:
EDIT: `static_cast` seems illegal here, changed to `reinterpret_cast` in https://github.com/NVIDIA/DALI/pull/3600/commits/ad25f59956ec68fedcd17f5bf265837d273f1a00